go-apps/meep-loc-serv/sbi/loc-serv-sbi.go

```diff
@@ -208,6 +234,139 @@ func processActiveScenarioUpdate() {
 	}
 }
 
+func processGisEngineUpdate(assetMap map[string]string) {
+	for assetName, assetType := range assetMap {
+		if assetType == postgis.TypeUe {
+			if assetName == postgis.AllAssets {
+				updateAllUserPosition()
+			} else {
+				updateUserPosition(assetName)
+			}
+		} else if assetType == postgis.TypePoa {
+			if assetName == postgis.AllAssets {
+				updateAllApPosition()
+			} else {
+				updateApPosition(assetName)
+			}
+		}
+	}
+}
+
+func getNetworkLocation(name string) (zone string, netLoc string, err error) {
+	ctx := sbi.activeModel.GetNodeContext(name)
+	if ctx == nil {
+		err = errors.New("Error getting context for: " + name)
+		return
+	}
+	nodeCtx, ok := ctx.(*mod.NodeContext)
+	if !ok {
+		err = errors.New("Error casting context for: " + name)
+		return
+	}
+	zone = nodeCtx.Parents[mod.Zone]
+	netLoc = nodeCtx.Parents[mod.NetLoc]
+	return zone, netLoc, nil
+}
+
+func parsePosition(position string) (longitude *float32, latitude *float32) {
+	var point dataModel.Point
+	err := json.Unmarshal([]byte(position), &point)
+	if err != nil {
+		return nil, nil
+	}
+	return &point.Coordinates[0], &point.Coordinates[1]
+}
+
+func updateUserPosition(name string) {
+	// Get network location
+	zone, netLoc, err := getNetworkLocation(name)
+	if err != nil {
+		log.Error(err.Error())
+		return
+	}
+
+	// Get position
+	var longitude *float32
+	var latitude *float32
+	ue, err := sbi.pc.GetUe(name)
+	if err == nil {
+		longitude, latitude = parsePosition(ue.Position)
+	}
+
+	// Update info
+	sbi.updateUserInfoCB(name, zone, netLoc, longitude, latitude)
+}
+
+func updateAllUserPosition() {
+	// Get all positions
+	ueMap, _ := sbi.pc.GetAllUe()
+
+	// Update info
+	ueNameList := sbi.activeModel.GetNodeNames("UE")
+	for _, name := range ueNameList {
+		// Get network location
+		zone, netLoc, err := getNetworkLocation(name)
+		if err != nil {
+			log.Error(err.Error())
+			return
+		}
+
+		// Get position
+		var longitude *float32
+		var latitude *float32
+		if ue, found := ueMap[name]; found {
+			longitude, latitude = parsePosition(ue.Position)
+		}
+
+		sbi.updateUserInfoCB(name, zone, netLoc, longitude, latitude)
+	}
+}
+
+func updateApPosition(name string) {
+	// Get network location
+	zone, netLoc, err := getNetworkLocation(name)
+	if err != nil {
+		log.Error(err.Error())
+		return
+	}
+
+	// Get position
+	var longitude *float32
+	var latitude *float32
+	poa, err := sbi.pc.GetPoa(name)
+	if err == nil {
+		longitude, latitude = parsePosition(poa.Position)
+	}
+
+	// Update info
+	sbi.updateAccessPointInfoCB(zone, netLoc, "UNKNOWN", "", -1, longitude, latitude)
+}
+
+func updateAllApPosition() {
```

Comment:
Same comment as above, I'd prefer to use the whole term AccessPoint, rather than Ap.. simply because it can be confused with App